app/models/project.rb

```diff
@@ -108,6 +110,28 @@ def user_can_add_observation?(obs, user)
                                is_member?(user))
   end
 
+  def violates_constraints?(obs)
+    violates_location?(obs) # || violates_date?(obs)
+  end
+
+  def violates_location?(obs)
+    return false if location.blank?
+
+    !location.found_here?(obs)
+  end
+
+  def count_violations
+    return 0 unless location
+
+    count = observations.where.not(lat: nil).count
+    count - observations.in_box(n: location.north, s: location.south,
+                                e: location.east, w: location.west).count
```

Comment:
If you're interested, I think there's a way to do this in Rails 6 in a single query, using Arel query that's kind of opaque.
(In Rails 7, it would be much clearer.)
```ruby
    observations.where(
      observations.in_box(n: location.north, s: location.south, 
                          e: location.east, w: location.west).
        arel.constraints.reduce(:and).not
    ).count
```
See https://makandracards.com/makandra/486959-how-to-negate-scope-conditions-in-rails

But unless `count_violations` gets called a lot, I like your code.